sonar-dotnet-tests-library/src/main/java/org/sonar/plugins/dotnet/tests/CoverageAggregator.java

```diff
@@ -110,8 +115,13 @@ Coverage aggregate(WildcardPatternFileProvider wildcardPatternFileProvider, Cove
   private void aggregate(WildcardPatternFileProvider wildcardPatternFileProvider, String[] reportPaths, CoverageParser parser, Coverage aggregatedCoverage) {
     for (String reportPathPattern : reportPaths) {
       if (!reportPathPattern.isEmpty()) {
-        for (File reportFile : wildcardPatternFileProvider.listFiles(reportPathPattern)) {
-          aggregatedCoverage.mergeWith(coverageCache.readCoverageFromCacheOrParse(parser, reportFile));
+        Set<File> filesMatchingPattern = wildcardPatternFileProvider.listFiles(reportPathPattern);
+        if (filesMatchingPattern.isEmpty()) {
+          LOG.warn("Could not find any coverage report file matching the pattern '{}'.", reportPathPattern);
+        } else {
+          for (File reportFile : filesMatchingPattern) {
+            aggregatedCoverage.mergeWith(coverageCache.readCoverageFromCacheOrParse(parser, reportFile));
```

Comment:
Testing here if the file exists is pointless because the `wildcardPatternFileProvider` already has filtered out files.